sdk/core/azure-core-amqp/vendor/azure-uamqp-c/src/link.c

```diff
@@ -801,11 +802,12 @@ LINK_HANDLE link_create(SESSION_HANDLE session, const char* name, role role, AMQ
                 }
                 else
                 {
+                    (void)memcpy(result->name, name, name_length + 1);
```

Comment:
Why was this move necessary? Could you add a comment explaining so that someone doesn't accidentally re-order it.

Reply:
The move was necessary because in C, it's important to initialize variables as soon as possible. Otherwise there is allocated but uninitialized memory that can potentially be accessed.

Reply:
After writing the comment explaining this change, I realized that it fell into the same category of comment as `foo = 1; // Assign 1 to foo`. And I really dislike that style of comments.